### PR TITLE
chore: upgrade to dbt core v1.4.0

### PR DESCRIPTION
**Pull request by nicor88**
### Description

This PR should cover this https://github.com/dbt-athena/dbt-athena/issues/136 also include python 3.11 to the matrix, doing so we know that we support the adapter with python 3.11

I tested with multiple models:
* parquet table
* parquet incremental and with partitions
* iceberg table
* iceberg append
* iceberg merge

# TODO
- [x] Fix functional testing - fixed -> https://github.com/dbt-labs/dbt-core/discussions/6624#discussioncomment-4863223
- [x] Investigate why caplog.text is empty

## Notes
- [ ] Somehow the caplog.text is not working anymore therefore tests are half broken

## Checklist
- [x] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [x] You added unit testing when necessary
- [x] You added functional testing when necessary

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [ "3.7", "3.8", "3.9", "3.10" ]
+        python-version: [ "3.7", "3.8", "3.9", "3.10", "3.11"]
```
Comment (Jrmyy) [inline .github/workflows/ci.yml:24]: Maybe add the final space to stay compliant with the beginning space ?

File: Makefile [hunk 2]
```diff
@@ -7,8 +7,8 @@ CHANGED_FILES_IN_BRANCH := $(shell git diff --name-only $(shell git merge-base o
 .PHONY : install_deps setup pre-commit pre-commit-in-branch pre-commit-all test help
 
 install_deps:  ## Install all dependencies.
-	pip install -r dev-requirements.txt
 	pip install -r requirements.txt
+	pip install -r dev-requirements.txt
```
Comment (Jrmyy) [inline Makefile:11]: Should we create one command `pip install -r requirements.txt -r dev-requirements.txt` ?

File: tests/unit/test_adapter.py [hunk 3]
```diff
@@ -182,17 +178,17 @@ def test_acquire_connection(self, connection_cls):
...
         conn_res = None
-        with pytest.raises(FailedToConnectException) as exc:
+        with pytest.raises(ConnectionError) as exc:
             conn_res = connection.handle
+
         assert conn_res is None
         assert connection.state == ConnectionState.FAIL
-        assert exc.value.msg == "foobar"
-        assert "Got an error when attempting to open a Athena connection due to foobar" in caplog.text
+        assert exc.value.__str__() == "foobar"
```
Comment (Jrmyy) [inline tests/unit/test_adapter.py:190]: `msg` property is not available anymore ?

**Review by Jrmyy — APPROVED** [review]
LGTM

Weird stuff about exceptions and logging, I might not see everything but it seems harder to use in 1.4 than in 1.3 🤔

File: tests/unit/test_adapter.py (hunk 3 above)
Comment (nicor88) [reply tests/unit/test_adapter.py:190]: Nope, but I found out that what we want is in the value.__str__

File: Makefile (hunk 2 above)
Comment (nicor88) [reply Makefile:11]: we can, I just had to change the order as I prefer to install core stuff lib first then the rest.

**Review by Jrmyy — APPROVED** [review]

**Pull request merged**